iam-login-service/src/main/java/it/infn/mw/iam/audit/IamAuthenticationAuditListener.java

```diff
@@ -0,0 +1,69 @@
+package it.infn.mw.iam.audit;
+
+import static it.infn.mw.iam.audit.IamAuditField.category;
+import static it.infn.mw.iam.audit.IamAuditField.details;
+import static it.infn.mw.iam.audit.IamAuditField.failureType;
+import static it.infn.mw.iam.audit.IamAuditField.generatedBy;
+import static it.infn.mw.iam.audit.IamAuditField.message;
+import static it.infn.mw.iam.audit.IamAuditField.principal;
+import static it.infn.mw.iam.audit.IamAuditField.source;
+import static it.infn.mw.iam.audit.IamAuditField.target;
+import static it.infn.mw.iam.audit.IamAuditField.type;
+import static it.infn.mw.iam.audit.IamAuditUtils.AUTHN_CATEGORY;
+
+import java.util.Map;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.context.ApplicationListener;
+import org.springframework.security.authentication.event.AbstractAuthenticationEvent;
+import org.springframework.security.authentication.event.AbstractAuthenticationFailureEvent;
+import org.springframework.security.authentication.event.InteractiveAuthenticationSuccessEvent;
+import org.springframework.security.web.authentication.switchuser.AuthenticationSwitchUserEvent;
+import org.springframework.stereotype.Component;
+
+import com.google.common.collect.Maps;
+
+@Component
+public class IamAuthenticationAuditListener
+    implements ApplicationListener<AbstractAuthenticationEvent> {
+
+  private static final Log logger = LogFactory.getLog(IamAuthenticationAuditListener.class);
+
+  private Map<String, Object> data = Maps.newLinkedHashMap();
+
+  @Override
+  public void onApplicationEvent(AbstractAuthenticationEvent event) {
+
+    data = Maps.newLinkedHashMap();
```

Comment:
Probably is more efficient to just clear the existing map, instead of allocating a new map for each event